### ISSUE TITLE
scheduler CrashLoopBackOff when there isn't enough sibling pods

### ISSUE THREAD

**Issue by zhiyxu** [kubernetes-sigs/scheduler-plugins#264]
I created a second scheduler in cluster via helm, create podgroup and deploy following installation doc, but when the scheduler cannot find enough sibling pods, it crashed:

```
# kubectl get pods | grep pause
pause-6688b6b475-j2n6v                              0/1     Pending            0          105s
pause-6688b6b475-qqwsp                              0/1     Pending            0          105s

# kubectl get pods -n scheduler-plugins 
NAME                                            READY   STATUS             RESTARTS   AGE
scheduler-plugins-controller-588f75f685-ztjf2   1/1     Running            0          57m
scheduler-plugins-scheduler-56f5cf9585-qcnp4    0/1     CrashLoopBackOff   5          18m

# kubectl logs scheduler-plugins-scheduler-56f5cf9585-qcnp4 -n scheduler-plugins 
I0927 07:48:29.290354       1 serving.go:331] Generated self-signed cert in-memory
W0927 07:48:29.914980       1 options.go:330] Neither --kubeconfig nor --master was specified. Using default API client. This might not work.
W0927 07:48:29.927419       1 client_config.go:614] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
W0927 07:48:30.029765       1 authorization.go:47] Authorization is disabled
W0927 07:48:30.029789       1 authentication.go:40] Authentication is disabled
I0927 07:48:30.029805       1 deprecated_insecure_serving.go:51] Serving healthz insecurely on 0.0.0.0:10251
I0927 07:48:30.032910       1 requestheader_controller.go:169] Starting RequestHeaderAuthRequestController
I0927 07:48:30.032918       1 configmap_cafile_content.go:202] Starting client-ca::kube-system::extension-apiserver-authentication::client-ca-file
I0927 07:48:30.032927       1 configmap_cafile_content.go:202] Starting client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file
I0927 07:48:30.032936       1 shared_informer.go:240] Waiting for caches to sync for client-ca::kube-system::extension-apiserver-authentication::client-ca-file
I0927 07:48:30.032936       1 shared_informer.go:240] Waiting for caches to sync for RequestHeaderAuthRequestController
I0927 07:48:30.032942       1 shared_informer.go:240] Waiting for caches to sync for client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file
I0927 07:48:30.035298       1 secure_serving.go:197] Serving securely on 0.0.0.0:10259
I0927 07:48:30.035307       1 tlsconfig.go:240] Starting DynamicServingCertificateController
I0927 07:48:30.133052       1 shared_informer.go:247] Caches are synced for client-ca::kube-system::extension-apiserver-authentication::client-ca-file 
I0927 07:48:30.133628       1 shared_informer.go:247] Caches are synced for client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file 
I0927 07:48:30.133781       1 shared_informer.go:247] Caches are synced for RequestHeaderAuthRequestController 
E0927 07:48:30.236026       1 coscheduling.go:137] pre-filter pod pause-6688b6b475-j2n6v cannot find enough sibling pods, current pods number: 2, minMember of group: 3
E0927 07:48:30.236502       1 runtime.go:78] Observed a panic: runtime.boundsError{x:0, y:0, signed:true, code:0x0} (runtime error: index out of range [0] with length 0)
goroutine 1 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1d88880, 0xc000f87240)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x89
panic(0x1d88880, 0xc000f87240)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption.pickOneNodeForPreemption(0xc000f8ab40, 0x2, 0x4)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:475 +0xcac
k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption.SelectCandidate(0xc001754480, 0x2, 0x4, 0xc0011b1510, 0x2160ae0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:415 +0x89
k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption.(*DefaultPreemption).preempt(0xc000439e00, 0x2160060, 0xc00033f580, 0xc000e789c0, 0xc0011b1510, 0xc000e78f90, 0xc04c7bb38e0d240c, 0x61c2302b, 0x2ea01a0, 0x501ff)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:152 +0x46d
k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption.(*DefaultPreemption).PostFilter(0xc000439e00, 0x2160060, 0xc00033f580, 0xc000e789c0, 0xc0011b1510, 0xc000e78f90, 0x0, 0x0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:98 +0xf1
k8s.io/kubernetes/pkg/scheduler/framework/runtime.(*frameworkImpl).runPostFilterPlugin(0xc0001b7d40, 0x2160060, 0xc00033f580, 0x7f645a1727a8, 0xc000439e00, 0xc000e789c0, 0xc0011b1510, 0xc000e78f90, 0xc000e45390, 0x14b4c17)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/framework/runtime/framework.go:592 +0x128
k8s.io/kubernetes/pkg/scheduler/framework/runtime.(*frameworkImpl).RunPostFilterPlugins(0xc0001b7d40, 0x2160060, 0xc00033f580, 0xc000e789c0, 0xc0011b1510, 0xc000e78f90, 0x0, 0x0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/framework/runtime/framework.go:574 +0x1e5
k8s.io/kubernetes/pkg/scheduler.(*Scheduler).scheduleOne(0xc00047e180, 0x2160060, 0xc00033f580)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/scheduler.go:479 +0x8a8
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000e45980)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000e45980, 0x2110f00, 0xc000e78960, 0xc00033f501, 0xc0003284e0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000e45980, 0x0, 0x0, 0x18d4e01, 0xc0003284e0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x2160060, 0xc00033f580, 0xc000e459e0, 0x0, 0x0, 0x8010101)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(...)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:99
k8s.io/kubernetes/pkg/scheduler.(*Scheduler).Run(0xc00047e180, 0x2160060, 0xc00033f580)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/scheduler.go:314 +0x92
k8s.io/kubernetes/cmd/kube-scheduler/app.Run(0x2160060, 0xc00033f580, 0xc00000e9c8, 0xc00047e180, 0x9, 0x9)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/cmd/kube-scheduler/app/server.go:220 +0x65b
k8s.io/kubernetes/cmd/kube-scheduler/app.runCommand(0xc0008a5b80, 0xc000314820, 0xc00033cc80, 0x9, 0x9, 0x0, 0x0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/cmd/kube-scheduler/app/server.go:132 +0x150
k8s.io/kubernetes/cmd/kube-scheduler/app.NewSchedulerCommand.func1(0xc0008a5b80, 0xc00033f540, 0x0, 0x4)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/cmd/kube-scheduler/app/server.go:81 +0x53
github.com/spf13/cobra.(*Command).execute(0xc0008a5b80, 0xc00004c0b0, 0x4, 0x4, 0xc0008a5b80, 0xc00004c0b0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/github.com/spf13/cobra/command.go:854 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc0008a5b80, 0x9, 0x9, 0xc0008a5b80)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/github.com/spf13/cobra/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/github.com/spf13/cobra/command.go:895
main.main()
	/go/src/sigs.k8s.io/scheduler-plugins/cmd/scheduler/main.go:66 +0x336
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x1d88880, 0xc000f87240)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption.pickOneNodeForPreemption(0xc000f8ab40, 0x2, 0x4)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:475 +0xcac
k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption.SelectCandidate(0xc001754480, 0x2, 0x4, 0xc0011b1510, 0x2160ae0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:415 +0x89
k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption.(*DefaultPreemption).preempt(0xc000439e00, 0x2160060, 0xc00033f580, 0xc000e789c0, 0xc0011b1510, 0xc000e78f90, 0xc04c7bb38e0d240c, 0x61c2302b, 0x2ea01a0, 0x501ff)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:152 +0x46d
k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption.(*DefaultPreemption).PostFilter(0xc000439e00, 0x2160060, 0xc00033f580, 0xc000e789c0, 0xc0011b1510, 0xc000e78f90, 0x0, 0x0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:98 +0xf1
k8s.io/kubernetes/pkg/scheduler/framework/runtime.(*frameworkImpl).runPostFilterPlugin(0xc0001b7d40, 0x2160060, 0xc00033f580, 0x7f645a1727a8, 0xc000439e00, 0xc000e789c0, 0xc0011b1510, 0xc000e78f90, 0xc000e45390, 0x14b4c17)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/framework/runtime/framework.go:592 +0x128
k8s.io/kubernetes/pkg/scheduler/framework/runtime.(*frameworkImpl).RunPostFilterPlugins(0xc0001b7d40, 0x2160060, 0xc00033f580, 0xc000e789c0, 0xc0011b1510, 0xc000e78f90, 0x0, 0x0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/framework/runtime/framework.go:574 +0x1e5
k8s.io/kubernetes/pkg/scheduler.(*Scheduler).scheduleOne(0xc00047e180, 0x2160060, 0xc00033f580)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/scheduler.go:479 +0x8a8
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000e45980)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000e45980, 0x2110f00, 0xc000e78960, 0xc00033f501, 0xc0003284e0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000e45980, 0x0, 0x0, 0x18d4e01, 0xc0003284e0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x2160060, 0xc00033f580, 0xc000e459e0, 0x0, 0x0, 0x8010101)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(...)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:99
k8s.io/kubernetes/pkg/scheduler.(*Scheduler).Run(0xc00047e180, 0x2160060, 0xc00033f580)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/pkg/scheduler/scheduler.go:314 +0x92
k8s.io/kubernetes/cmd/kube-scheduler/app.Run(0x2160060, 0xc00033f580, 0xc00000e9c8, 0xc00047e180, 0x9, 0x9)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/cmd/kube-scheduler/app/server.go:220 +0x65b
k8s.io/kubernetes/cmd/kube-scheduler/app.runCommand(0xc0008a5b80, 0xc000314820, 0xc00033cc80, 0x9, 0x9, 0x0, 0x0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/cmd/kube-scheduler/app/server.go:132 +0x150
k8s.io/kubernetes/cmd/kube-scheduler/app.NewSchedulerCommand.func1(0xc0008a5b80, 0xc00033f540, 0x0, 0x4)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/k8s.io/kubernetes/cmd/kube-scheduler/app/server.go:81 +0x53
github.com/spf13/cobra.(*Command).execute(0xc0008a5b80, 0xc00004c0b0, 0x4, 0x4, 0xc0008a5b80, 0xc00004c0b0)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/github.com/spf13/cobra/command.go:854 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc0008a5b80, 0x9, 0x9, 0xc0008a5b80)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/github.com/spf13/cobra/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/github.com/spf13/cobra/command.go:895
main.main()
	/go/src/sigs.k8s.io/scheduler-plugins/cmd/scheduler/main.go:66 +0x336
```

version: v0.20.10

**Comment by denkensk** [comment]
It crashed from here in default preempt. I can help to check it tomorrow.
https://github.com/kubernetes/kubernetes/blob/8152330a2b6ca3621196e62966ef761b8f5a61bb/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go#L475

**Comment by Huang-Wei** [comment]
@zhiyxu I cannot reproduce this in a brand new kind cluster. Are you running some other in-house extensions, like extenders? If yes, it's also reported in https://github.com/kubernetes/kubernetes/issues/101548 and we had a fix in 1.22.

**Comment by zhiyxu** [comment]
> @zhiyxu I cannot reproduce this in a brand new kind cluster. Are you running some other in-house extensions, like extenders? If yes, it's also reported in [kubernetes/kubernetes#101548](https://github.com/kubernetes/kubernetes/issues/101548) and we had a fix in 1.22.

thanks, I'll try to test in cluster of latest version.
